UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -37,7 +28,6 @@ public class Character : MonoBehaviour
     private void Awake()
     {
         characterController = GetComponent<CharacterController>();
```

Comment:
You need to cache `characterAnimator` here.
```suggestion
        characterController = GetComponent<CharacterController>();
        characterAnimator= GetComponent<CharacterAnimator >();
```